### PR TITLE
Issue with refobject

### LINKED ISSUE #2: Issue with type of RefObject<HTMLElement>
# Issue with type of RefObject<HTMLElement>

Screenshot 2022-07-13 at 2 15 11 AM

## Can we accept this as a solution?

```js
type ReturnArr = [ref: RefObject<any>, inView: boolean];
```

I couldn't figure out what anything other than `any` that would be one of the right fit here.

Can anyone suggest? If you think any is fine. Then i can create an PR as per that.

### PR DESCRIPTION
**Pull request by HenrikDK2**
Fixes

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -1,34 +1,46 @@
...
   triggerOnce?: boolean;
 }
 
-type ReturnArr = [ref: RefObject<HTMLElement>, inView: boolean];
 const isBrowser = typeof window !== "undefined";
 
-export default function (options: ObserverOptions): ReturnArr {
-  const ref: RefObject<HTMLElement> = createRef();
-  const init: RefObject<boolean> = createRef();
+export const useObserver = <T extends HTMLElement>(
```
Comment (samarpanda) [inline src/index.tsx:12]: This introduced a breaking change as it changed the signature i.e 

```js
// Earlier
import useObserver from 'preact-intersection-observer';

// Now
import { useObserver } from 'preact-intersection-observer';
````
As the export default got removed. Hope it was intentional?

Comment (samarpanda) [reply src/index.tsx:12]: Although no problems for me. As i have already changed it accordingly. 

But the guys who would be using this package already in any project. They have to change once they accept the minor version upgrade.

**Comment by samarpanda** [comment]
Thanks for addressing it pretty quick. 👍👏

File: src/index.tsx (hunk 1 above)
Comment (HenrikDK2) [reply src/index.tsx:12]: This was intentional from my side, but you're right I only made a minor patch. I will add a small bugfix that includes exported default as optional way of importing the module.

**Pull request merged**